hack/gen-crd.sh

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env bash
+
```

Comment:
can't we just modify the `pkg/apis` layout here to satisfy the crd command then just:
```
go run controller-gen/main.go crd --domain openshift.io --output-dir=~/my-new-crds/
pushd cluster-api
go run controller-gen/main.go crd --domain openshift.io --output-dir=~/my-new-crds/
```

Reply:
```
$ pwd
/home/jchaloup/Projects/src/github.com/openshift/machine-api-operator/vendor/github.com/openshift/cluster-api
$ go run ../../../sigs.k8s.io/controller-tools/cmd/controller-gen/main.go crd --domain openshift.io
W0403 10:13:42.055714   16847 parse.go:224] Ignoring directory ./pkg/apis: No files for pkg "github.com/openshift/cluster-api/pkg/apis"
CRD files generated, files can be found under path /home/jchaloup/Projects/src/github.com/openshift/machine-api-operator/vendor/github.com/openshift/cluster-api/config/crds.
$ ls /home/jchaloup/Projects/src/github.com/openshift/machine-api-operator/vendor/github.com/openshift/cluster-api/config/crds
$ 
```

No CRDs got generated.

Reply:
:/ thanks for trying it! I still don't wanna give up on finding a nicer way, let us investigate a bit more before merging